skorecard/reporting/report.py

```diff
@@ -123,7 +123,10 @@ def build_bucket_table(
     stats["% Event"] = stats["Event"] / stats["Event"].sum()
     stats["% Non-event"] = stats["Non-event"] / stats["Non-event"].sum()
 
-    stats["WoE"] = ((stats["% Non-event"] + epsilon) / (stats["% Event"] + epsilon)).apply(lambda x: np.log(x))
+    event_percentage = (stats["Event"] + epsilon) / (stats["Event"].sum() + 2 * epsilon)
+    non_event_percentage = (stats["Non-event"] + epsilon) / (stats["Non-event"].sum() + 2 * epsilon)
+    stats["WoE"] = (event_percentage / non_event_percentage).apply(lambda x: np.log(x))
```

Comment:
The WoE is normally calculated in the following function

```python
from skorecard.metrics.metrics import woe_1d
```

and then this is propagated through calculations (see the `WoEEncoder`). 
This should guarantee consistency across the package. I would suggest using it in here

Reply:
Hi Sandro, thank you very much for your comments, I am glad to see you are still looking after the project.
 
I was under the impression that WoEEncoder was deprecated (line 18 of _WoEencoder.py) in favor of category_encoders WOEEncoder , which is the one in use in the Skorecard class (line 216 of skorecard.py).
This is also why I set epsilon=1: I agree with you that it makes a lot more sense to have it small, but skorecard.py uses the default value for category_encoders' WOEEncoder, namely "regularisation=1" (line 77 of  
[https://github.com/scikit-learn-contrib/category_encoders/blob/master/category_encoders/woe.py](url)
).  

Would it be your suggestion to continue working with category_encoder's WOEEncoder, or with skorecard's WoEEncoder? If continuing with category_encoders' WOEEncoder, a remedy for the regularisation would be to set it back to 10**(-5) in bucket_table(), but then to also alter the call to WOEEncoder in the skorecard.py.

Reply:
Hi @dlaprins, you are correct. Indeed the epsilon is added at denominator and numerator level, not in the ratio.
And after checking a bit more, I noticed that I missed this deprecation of WoEEncoder that happened in the issue #53.

As this method is being deprecated, it makes sense to keep the change you suggest in this PR.